network/simulation/simulation.go

```diff
@@ -76,8 +85,44 @@ func New(services map[string]ServiceFunc) (s *Simulation) {
 		buckets:           make(map[enode.ID]*sync.Map),
 		done:              make(chan struct{}),
 		neighbourhoodSize: network.NewKadParams().NeighbourhoodSize,
+		typ:               SimulationTypeInproc,
+	}
+
+	adapterServices := addServices(s, services)
+
+	s.Net = simulations.NewNetwork(
+		adapters.NewTCPAdapter(adapterServices),
+		&simulations.NetworkConfig{ID: "0"},
+	)
+
+	return s
+}
+
+// NewExec does the same as New but lets the caller specify the adapter to use
+func NewExec(services map[string]ServiceFunc) (s *Simulation, err error) {
```

Comment:
Probably worth renaming `New` to `NewInProc`, since we are adding a new `NewExec` constructor?